cyclops-ui/src/components/pages/Modules/Modules.tsx

```diff
@@ -45,11 +49,40 @@ const Modules = () => {
         setLoadingModules(false);
       });
   }, []);
+  useEffect(() => {
+    const newfilteredData = filteredData.filter((module: any) =>
+      moduleHealthFilter
+        .map((status) => status.toLowerCase())
+        .includes(module.status.toLowerCase()),
+    );
+
+    if (JSON.stringify(newfilteredData) !== JSON.stringify(filteredData)) {
+      setFilteredData(newfilteredData);
+    }
+  }, [moduleHealthFilter, filteredData]);
```

Comment:
I tried it locally, and it doesn't seem to work. The problem is that when you unselect one health filter and select it again, it will not show the modules with that health status. This happens because you are always filtering on top of existing `filteredData`, which doesn't contain all modules but only the ones that passed the previous filter.

I would propose a different solution. You can have a similar `useEffect` that has `moduleHealthFilter` as a dependency and the input to the search field as a dependency. This will require you to store the search input text as a separate state (you can call this other state `searchInputFilter`), and the `handleSearch` function can simply store the search input in the state.

When you have both `moduleHealthFilter` and `searchInputFilter` (search input text), you can filter based on the health as you did above and then apply the filter like it's implemented in the `handleSearch` now. After filtering, you can simply store it in the `filteredData`.

Reply:
I have tried something can you please check once now it would be great

Reply:
@nikhiltanna, you added `allData` as a dependency, but this way, we are performing filters in separate functions. In the solution suggested above, filters are applied in the same place, so it's easier to read and maintain.

Reply:
I made the changes as asked and thanks for the suggestion it really makes code readable as all filter functions gather into one place